### PR TITLE
fix: type error in `UserDisplay` component example

### PR DESCRIPTION
**Pull request by MikePrus**
There is a type error in the `UserDisplay` component example because we don't display the name but the object instead.

### CODE REVIEW

**Comment by markerikson** [comment]
Hi! In this case the example is actually correct.  If you look at the `WritingTests.mdx` file, there's a "hidden" `userSlice.ts` file defined in the code snippet, with these lines:

```ts
export const selectUser = (state: RootState) => state.user.name
export const selectUserFetchStatus = (state: RootState) => state.user.status
export default userSlice.reducer
```

So, it _is_ getting the `user.name` string already.

Could you change the selector name to be `selectUserName` and the variable name to be `userName` instead?

**Comment by MikePrus** [comment]
> Hi! In this case the example is actually correct. If you look at the `WritingTests.mdx` file, there's a "hidden" `userSlice.ts` file defined in the code snippet, with these lines:
> 
> ```ts
> export const selectUser = (state: RootState) => state.user.name
> export const selectUserFetchStatus = (state: RootState) => state.user.status
> export default userSlice.reducer
> ```
> 
> So, it _is_ getting the `user.name` string already.
> 
> Could you change the selector name to be `selectUserName` and the variable name to be `userName` instead?

Sure, I did this in the second commit please review.

File: docs/usage/WritingTests.mdx [hunk 1]
```diff
@@ -563,17 +563,17 @@ export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
...
   const dispatch = useAppDispatch()
-  const user = useAppSelector(selectUser)
+  const user = useAppSelector(selectUserName)
   const userFetchStatus = useAppSelector(selectUserFetchStatus)
 
   return (
     <div>
       {/* Display the current user name */}
-      <div>{user}</div>
+      <div>{user.name}</div>
```
Comment (markerikson) [inline docs/usage/WritingTests.mdx:576]:
```suggestion
      <div>{userName}</div>
```

File: docs/usage/WritingTests.mdx [hunk 2]
```diff
@@ -563,17 +563,17 @@ export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
 // file: features/users/UserDisplay.tsx noEmit
 import React from 'react'
 import { useAppDispatch, useAppSelector } from '../../app/hooks'
-import { fetchUser, selectUser, selectUserFetchStatus } from './userSlice'
+import { fetchUser, selectUserName, selectUserFetchStatus } from './userSlice'
 
 export default function UserDisplay() {
   const dispatch = useAppDispatch()
-  const user = useAppSelector(selectUser)
+  const user = useAppSelector(selectUserName)
```
Comment (markerikson) [inline docs/usage/WritingTests.mdx:570]:
```suggestion
  const userName = useAppSelector(selectUserName)
```

**Review by markerikson — COMMENTED** [review]
Fix userName naming

**Comment by vincentCheng** [comment]
I also found the error.

**Pull request merged**